source/Body.cpp

```diff
@@ -205,90 +251,399 @@ const Government *Body::GetGovernment() const
+		else if(child.Token(0) == "center" && child.Size() >= 3)
+			spriteAnimationParameters.center = Point(child.Value(1), child.Value(2));
+		else if(child.Token(0) == "frame rate" && child.Size() >= 2 && child.Value(1) >= 0.)
+			spriteAnimationParameters.frameRate = child.Value(1) / 60.;
 		else if(child.Token(0) == "frame time" && child.Size() >= 2 && child.Value(1) > 0.)
-			frameRate = 1. / child.Value(1);
+			spriteAnimationParameters.frameRate = 1. / child.Value(1);
 		else if(child.Token(0) == "delay" && child.Size() >= 2 && child.Value(1) > 0.)
-			delay = child.Value(1);
+			spriteAnimationParameters.delay = child.Value(1);
+		else if(child.Token(0) == "transition delay" && child.Size() >= 2 && child.Value(1) >= 0.)
+			spriteAnimationParameters.transitionDelay = child.Value(1);
 		else if(child.Token(0) == "scale" && child.Size() >= 2 && child.Value(1) > 0.)
-			scale = static_cast<float>(child.Value(1));
+			spriteAnimationParameters.scale = static_cast<float>(child.Value(1));
 		else if(child.Token(0) == "start frame" && child.Size() >= 2)
 		{
-			frameOffset += static_cast<float>(child.Value(1));
-			startAtZero = true;
+			spriteAnimationParameters.startFrame = static_cast<float>(child.Value(1));
+			spriteAnimationParameters.startAtZero = true;
+		}
+		else if(child.Token(0) == "ramp" && child.Size() >= 2)
+		{
+			if(child.Size() == 2)
+			{
+				spriteAnimationParameters.rampUpRate = child.Value(1) / 3600.;
+				spriteAnimationParameters.rampDownRate = child.Value(1) / 3600.;
+			}
+			else if(child.Size() == 3)
+			{
+				spriteAnimationParameters.rampUpRate = child.Value(1) / 3600.;
+				spriteAnimationParameters.rampDownRate = child.Value(2) / 3600.;
+			}
+		}
+		else if(child.Token(0) == "indicate percentage" && child.Size() >= 2 && child.Value(1) > 0. && child.Value(1) < 1.)
+		{
+			spriteAnimationParameters.indicateReady = true;
+			spriteAnimationParameters.repeat = false;
+			spriteAnimationParameters.startAtZero = true;
+			spriteAnimationParameters.indicatePercentage = static_cast<float>(child.Value(1));
+		}
+		else if(child.Token(0) == "no indicate")
+			spriteAnimationParameters.indicateReady = false;
+		else if(child.Token(0) == "indicate")
+		{
+			spriteAnimationParameters.indicateReady = true;
+			spriteAnimationParameters.repeat = false;
+			spriteAnimationParameters.startAtZero = true;
+			spriteAnimationParameters.indicatePercentage = 1.0f;
 		}
+		else if(child.Token(0) == "random")
+			spriteAnimationParameters.randomize = true;
 		else if(child.Token(0) == "random start frame")
-			randomize = true;
+			spriteAnimationParameters.randomizeStart = true;
 		else if(child.Token(0) == "no repeat")
 		{
-			repeat = false;
-			startAtZero = true;
+			spriteAnimationParameters.repeat = false;
+			spriteAnimationParameters.startAtZero = true;
+		}
+		else if(child.Token(0) == "transition rewind")
+		{
+			spriteAnimationParameters.transitionFinish = false;
+			spriteAnimationParameters.transitionRewind = true;
```

Comment:
They are not necessarily mutually exclusive, there is a case where both of them are false where the sprite animation behaves differently. However, now that you point this out, I think this can be better designed i.e. have one input `transition type` and have a defined enumeration:

- `transition type 0`: Transition immediately into the next state
- `transition type 1`: Transition after finishing the current animation; same as `transitionFinish`
- `transition type 2`: Transition after rewinding the current animation; same as `transitionRewind`

This could also make it easier for newer transition types to be added in the future.